### DISCUSSION TITLE
Client-server network architecture

### DISCUSSION THREAD

**Discussion by electrofloat** [fail2ban/fail2ban#3482] [Q&A] [answered]
Hi!

As far as I know while in fail2ban there exists a fail2ban-client and fail2ban-server, they both have to be on the same host.

Are there/were there any plans on moving this single host architecture to a multi host one?

An example where this could be beneficial:
- There are multiple hosts, with multiple services, each logging on the host itself, but the ban itself has to be on a different host. (now, one could just do away with sending the logs to the host where the ban should take place, but let's assume that is not feasible.)

**Comment by sebres** [comment, marked as answer]
fail2ban-server and fail2ban-client has nothing to do with a network architecture: server is intended to implement the service process (running from systemd unit, init.d script, etc), client is used to communicate with server (reload, (re)start, stop, ban/unban etc).

Regarding the subject - here is related RFE #881
